.github/workflows/gradle-check.yml

```diff
@@ -16,58 +16,33 @@ jobs:
             echo "pr_title=$(jq --raw-output .pull_request.title $GITHUB_EVENT_PATH)" >> $GITHUB_ENV
             echo "pr_number=$(jq --raw-output .pull_request.number $GITHUB_EVENT_PATH)" >> $GITHUB_ENV
 
-      - name: Trigger jenkins workflow to run gradle check
-        run: |
-            echo "${{ env.pr_to_clone_url }} PR: ${{ env.pr_number }}"
-            echo "Raise from ${{ env.pr_from_clone_url }} ${{ env.pr_from_sha }}"
-            echo "PR Title: ${{ env.pr_title }}"
-            JENKINS_URL="https://build.ci.opensearch.org"
-            TIMEPASS=0
-            TIMEOUT=7200
-            RESULT="null"
-
-            echo "Trigger Jenkins workflows"
-            JENKINS_REQ=$(curl -s -XPOST \
-                 -H "Authorization: Bearer ${{ secrets.JENKINS_GRADLE_CHECK_GENERIC_WEBHOOK_TOKEN }}" \
-                 -H "Content-Type: application/json" \
-                 --data '{"pr_from_sha": "${{ env.pr_from_sha }}", "pr_from_clone_url": "${{ env.pr_from_clone_url }}", "pr_to_clone_url": "${{ env.pr_to_clone_url }}", "pr_title": "${{ env.pr_title }}", "pr_number": "${{ env.pr_number }}"}' \
-                 'https://build.ci.opensearch.org/generic-webhook-trigger/invoke')
-
-            QUEUE_URL=$(echo $JENKINS_REQ | jq --raw-output '.jobs."gradle-check".url')
-            echo QUEUE_URL $QUEUE_URL
-            echo "wait for jenkins to start workflow" && sleep 15
-
-            echo "Check if queue exist in Jenkins after triggering"
-            if [ -z "$QUEUE_URL" ] || [ "$QUEUE_URL" != "null" ]; then
-                WORKFLOW_URL=$(curl -s -XGET ${JENKINS_URL}/${QUEUE_URL}api/json | jq --raw-output .executable.url)
-                echo WORKFLOW_URL $WORKFLOW_URL
-
-                echo "Use queue information to find build number in Jenkins if available"
-                if [ -z "$WORKFLOW_URL" ] || [ "$WORKFLOW_URL" != "null" ]; then
+      - name: Checkout opensearch-build repo
+        uses: actions/checkout@v2
+        with:
+          repository: opensearch-project/opensearch-build
```

Comment:
Instead of having dependency on the build repo. Can we do something suggested here: https://github.com/opensearch-project/OpenSearch/pull/3717#issuecomment-1168899170?

Reply:
Nope, that workflow have not been maintained for over 3 years and it is in 0.0.1 version.
I will not use that for production purposes.

Thanks.

Reply:
Also that one is for API token not the same as generic webhook, where we can pass a lot of information between github and Jenkins.

I dont see a dependency on opensearch-build is an issue the jenkinsfile is in opensearch-build already.

Reply:
Can we have the scripts present in OpenSearch repo? Also, a good practice to mark the conversation resolved by the reviewer can be found [here](https://github.com/distributed-system-analysis/pbench/discussions/2113)

Reply:
The point to present it in opensearch-build repo is so that you dont need to backport to every single branch whenever there is a tiny change.

Remember you have more than 10+ branches need gradle check.

Reply:
Thanks for raising this issues tho, but I believe this is the best way to get gradle check start running.
We can always improve in later PRs.

Thanks.

Reply:
@peterzhuamazon : Thanks for your effort in pulling up this change. 

Regarding location of gradle script, I think it logically belongs to OpenSearch repository. Once the change is stabilized (green runs on main), we can backport this to remaining branches. Going forward there wouldn't be work needed for future branches cut from main (or release branches). 

@dblock WDYT ?

Reply:
@dreamer-89 that's a good point, I see both (central repo for build scripts and replicating build scripts across many branches). I think having build material along each branch is the best option when build scripts are trivial or non-existent (like pure Github actions let say). When build scripts are non-trivial (I think this is our case at least now), keeping them centralized at some place (build repository / library) serves better.

Reply:
Hi @reta @dreamer-89 

Yes that is the reason why I keep the script in build repo for now.
All the existing actions on market are either not meeting our needs, or too old without any maintenance.
We can improve this later by either forking one of the existing actions, or write a new one to replace.

Right now, like you suggested, it is best to leave it in build repo for now.

Thanks.